nixos/tests/acme.nix

```diff
@@ -266,6 +266,36 @@ in {
           }
         ];
 
+        concurrency-limit.configuration = { pkgs, ... }: lib.mkMerge [
+          webserverBasicConfig
+          {
+            security.acme.maxConcurrentRenewals = 1;
+
+            services.nginx.virtualHosts = {
+              # To trigger renew on a.example.test...
+              "a.example.test".serverAliases = [ "e.example.test" ];
+              "f.example.test" = vhostBase // {
+                enableACME = true;
+              };
+              "g.example.test" = vhostBase // {
+                enableACME = true;
+              };
+            };
+
+            systemd.services = {
+              "acme-a.example.test".serviceConfig.ExecStartPre = "+" + (pkgs.writeShellScript "test-a" ''
+                test "$(systemctl is-active acme-{f,g}.example.test.service | grep activating | wc -l)" -le 0
```

Comment:
Clever approach of checking the mutual exclusion.
I wonder whether it makes sense to check with `maxConcurrentRenewals > 1` as well, as the possibility of allowing a certain amount of parallel runs is what makes this issue rather complex. But checking whether this parallelism actually happens might be too cumbersome in test VM environments, making your existing test at least a good smoke test to notice when limiting breaks at all.